src/tree/containers/ContainerTreeItem.ts

```diff
@@ -87,4 +96,36 @@ export class ContainerTreeItem extends AzExtTreeItemIntermediate {
     public async deleteTreeItemImpl(context: IActionContext): Promise<void> {
         return ext.dockerClient.removeContainer(context, this.containerId);
     }
+
+    public hasMoreChildrenImpl(): boolean {
+        return this._item.showFiles && this.isRunning && this.children === undefined;
+    }
+
+    public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        if (clearCache) {
+            this.children = undefined;
+        }
+
+        if (this._item.showFiles && this.isRunning) {
+            this.children = [
+                new FilesTreeItem(
+                    this,
+                    vscode.workspace.fs,
+                    this.containerId,
+                    async c => {
```

Comment:
This could also be replaced with an `AsyncLazy` which would accomplish more or less the same thing.

Reply:
The need (i.e. desire) to pass along the `IActionContext` that bubbles up from the actual user interaction that necessitates inspecting the container means that we can't use `AsyncLazy<T>`.  (For what it's worth, I feel like `IActionContext` is leaking its way into too many of the interactions and APIs across the codebase, mostly in order to pass it along to an eventual Docker API call.  I wonder if we could make it more of an ambient property that flows through each asynchronous user interaction to be extracted at those points that need it.  I *think* Node has a mechanism for this (what we'd call "thread local storage" in .NET.)